### PR TITLE
Warn when a user creates a new project inside a exisiting git repo

### LINKED ISSUE #487: loco-cli should warn when trying to create an app inside another git repository
## Feature Request

Every developer did it at least once in their life - starting a new project (`npm init`/`loco new`) inside an already existing project, then realizing it too late and starting moving things.

**Describe the solution you'd like**

I would like the loco-cli to warn the user when running the command if there is a `.git` folder up in the directory tree and ask them explicitly to allow that.

**Describe alternatives you've considered**

1st alternative: As of today, ignore that, which I obviously don't think it's a good idea.
2nd alternative: Don't allow that at all. I think this alternative is too strict to the other side when sometimes the user does want to do that.
  - This can be done by allowing an explicit flag to the command line, but because it's already interactive, why not keep it interactive?

As a side note, I would love to help develop this feature request with appropriate mentoring.

### PR DESCRIPTION
**Pull request by assapir**
Closes #487.
I tested it manually, I couldn't find any automated tests for the CLI

### CODE REVIEW

File: loco-cli/src/prompt.rs [hunk 1]
```diff
@@ -78,6 +78,36 @@ pub fn template_selection(
...
+
+    let res = requestty::prompt_one(question)?;
+    let answer = res
+        .as_bool()
+        .ok_or_else(|| eyre::eyre!("allow_git_repo is empty"))?;
+
+    if answer {
+        Ok(())
+    } else {
+        Err(eyre::eyre!("user choose not to continue"))
```
Comment (jondot) [inline loco-cli/src/prompt.rs:107]: Maybe a more personal "aborted: you've chosen to quit."

**Review by jondot — APPROVED** [review]

File: loco-cli/src/git.rs [hunk 2]
```diff
@@ -144,6 +144,23 @@ fn git_exists() -> bool {
...
+pub fn is_a_git_repo(destination_path: &PathBuf) -> eyre::Result<bool> {
+    let destination_path = destination_path.canonicalize()?;
+    match Command::new("git")
+        .arg("-C")
+        .arg(destination_path)
+        .arg("rev-parse")
+        .arg("--is-inside-work-tree")
+        .output()
+    {
+        Ok(output) => Ok(output.status.success()),
```
Comment (kaplanelad) [inline loco-cli/src/git.rs:156]: let logged the error if the command returned an error
```suggestion
     Ok(output) => {
            if output.status.success() {
                Ok(true)
            } else {
                tracing::error!(
                    error = tracing::field::debug(output),
                    "git command returned an error"
                );
                Ok(false)
            }
        }
```

File: loco-cli/src/prompt.rs [hunk 3]
```diff
@@ -78,6 +78,36 @@ pub fn template_selection(
...
+/// If the `ALLOW_IN_GIT_REPO` environment variable is set, this test will be skipped.
+/// If the environment variable is not set, the function will warn the user if they are inside a git
+/// and will let them cancel the operation or continue.
+///
+/// # Errors
+/// when could not prompt the question, or when the user choose not to continue.
+pub fn warn_if_in_git_repo() -> eyre::Result<()> {
+    if let Ok(_) = env::var("ALLOW_IN_GIT_REPO") {
+        return Ok(());
+    }
```
Comment (kaplanelad) [inline loco-cli/src/prompt.rs:92]:
```suggestion
    if env::var("ALLOW_IN_GIT_REPO").is_ok() {
        return Ok(());
    }
```

File: loco-cli/src/git.rs (hunk 2 above)
Comment (kaplanelad) [reply loco-cli/src/git.rs:156]: Id the command not succeeded let print the strerr to log.

**Review by kaplanelad — COMMENTED** [review]
Hey @assapir good to see you here :)!
See small comments

**Comment by assapir** [comment]
> Hey @assapir good to see you here :)! See small comments

Thanks @kaplanelad, I said I would open PRs, so I did :)
Fixed regarding your comments, please take a look again.

**Review by kaplanelad — APPROVED** [review]

**Comment by kaplanelad** [comment]
@thanks @assapir you are man of word.
i'll publish the cli later on today

**Comment by assapir** [comment]
> @thanks @assapir you are man of word. i'll publish the cli later on today

Do you want me to create a GH action for automating that? if so, just create a secret with the creds needed and I will add that :)

**Pull request merged**